wagtail/tests/customuser/models.py

```diff
@@ -45,3 +49,61 @@ def get_full_name(self):
 
     def get_short_name(self):
         return self.first_name
+
+
+class EmailUserManager(BaseUserManager):
+    def _create_user(self, email, password,
+                     is_staff, is_superuser, **extra_fields):
+        """
+        Creates and saves a User with the given email and password.
+        """
+        email = self.normalize_email(email)
+        user = self.model(email=email, is_staff=is_staff, is_active=True,
+                          is_superuser=is_superuser, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email=None, password=None, **extra_fields):
+        return self._create_user(email, password, False, False,
+                                 **extra_fields)
+
+    def create_superuser(self, email, password, **extra_fields):
+        return self._create_user(email, password, True, True,
+                                 **extra_fields)
+
+
+class EmailUser(AbstractBaseUser):
+    # Cant inherit from PermissionsMixin because of clashes with
+    # groups/user_permissions related_names.
+    email = models.EmailField(max_length=255, unique=True)
+    is_staff = models.BooleanField(default=True)
+    is_active = models.BooleanField(default=True)
+    first_name = models.CharField(max_length=50, blank=True)
+    last_name = models.CharField(max_length=50, blank=True)
+
+    is_superuser = models.BooleanField(default=False)
```

Comment:
If you tell Django that this is swappable by adding:
```
class Meta:
    swappable = 'AUTH_USER_MODEL'
```

That'll make Django let you use the ``PermissionsMixin`` without the ``related_name`` conflicts